addons/mail/models/mail_message.py

```diff
@@ -1171,10 +1171,17 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
+        messages_per_partner = defaultdict(lambda: self.env['mail.message'])
+        for message in messages:
+            partner = self.env.user.partner_id
+            author = message.author_id
+            messages_per_partner[author] |= message
+            if author != partner:
+                messages_per_partner[partner] |= message
```

Comment:
This can more easily be written as :
```suggestion
            messages_per_partner[self.env.user.partner_id] |= message
            messages_per_partner[message.author_id] |= message
```
But actually `author_id` is not guaranteed to be set, and currently user might actually be a public user (livechat), so it would be safer to guard them:
```suggestion
            if not self.env.user._is_public():
                messages_per_partner[self.env.user.partner_id] |= message
            if message.author_id:
                messages_per_partner[message.author_id] |= message
```